### ISSUE TITLE
Error using vertcat: Dimensions of matrices being concatenated are not consistent.

### ISSUE THREAD

**Issue by daa233** [cvjena/part_constellation_models#2]
Hi,
I'm trying to run the code with your pre-trained models. I use the following command in MATLAB terminal:
```MATLAB
start('cache_dir', 'Simon_ICCV15_models/cache_iccv_cub200', ...
      'cnn_dir', 'Simon_ICCV15_models/cnn_finetuning/vgg19/',...
      'crop_size', 224);
```
Then I got the output and an error like:
```
opts = 

  struct with fields:

                  caffe_path: './lib/caffe_pp'
              liblinear_path: './lib/liblinear-2.1/matlab/'
              selsearch_path: './lib/SelectiveSearchCodeIJCV/'
                    imagedir: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/images'
                     basedir: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011'
                   cache_dir: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cache/'
              store_features: 0
                     cnn_dir: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cnn_finetuning/caffe_reference/'
                  batch_size: 11
                   crop_size: 227
                  finetuning: 1
            finetuning_iters: 10000
              finetuning_gpu: 0
               estimate_bbox: 1
              calculate_mean: 1
                  part_layer: 'pool5'
    part_layer_channel_count: 256
           no_selected_parts: 10
            no_visible_parts: 5
                  view_count: 10
                  iterations: 5
                  class_wise: 1
              feature_global: 'relu7'
                 use_flipped: 1
            use_bounding_box: 1
              pyramid_levels: 0
              rand_tr_images: -1
                rand_tr_part: 0
                 repetitions: 1
                   use_parts: 1
                 part_scales: [0.4400 0.2400]
                feature_part: 'relu7'
                  svm_params: '-q'
              parfor_workers: 2
                  use_parfor: 1
              verbose_output: 1
              imagelist_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/images.txt'
                  tr_ID_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/train_test_split.txt'


opts = 

  struct with fields:

                  caffe_path: './lib/caffe_pp'
              liblinear_path: './lib/liblinear-2.1/matlab/'
              selsearch_path: './lib/SelectiveSearchCodeIJCV/'
                    imagedir: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/images'
                     basedir: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011'
                   cache_dir: 'Simon_ICCV15_models/cache_iccv_cub200'
              store_features: 0
                     cnn_dir: 'Simon_ICCV15_models/cnn_finetuning/vgg19/'
                  batch_size: 11
                   crop_size: 224
                  finetuning: 1
            finetuning_iters: 10000
              finetuning_gpu: 0
               estimate_bbox: 1
              calculate_mean: 1
                  part_layer: 'pool5'
    part_layer_channel_count: 256
           no_selected_parts: 10
            no_visible_parts: 5
                  view_count: 10
                  iterations: 5
                  class_wise: 1
              feature_global: 'relu7'
                 use_flipped: 1
            use_bounding_box: 1
              pyramid_levels: 0
              rand_tr_images: -1
                rand_tr_part: 0
                 repetitions: 1
                   use_parts: 1
                 part_scales: [0.4400 0.2400]
                feature_part: 'relu7'
                  svm_params: '-q'
              parfor_workers: 2
                  use_parfor: 1
              verbose_output: 1
              imagelist_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/images.txt'
                  tr_ID_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/train_test_split.txt'
                 labels_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/image_class_labels.txt'
                      deploy: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cnn_finetuning/caffe_reference//deploy.prototxt'
                       model: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cnn_finetuning/caffe_reference//model'
               mean_mat_file: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cnn_finetuning/caffe_reference//mean.mat'
             mean_proto_file: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cnn_finetuning/caffe_reference//mean.binaryproto'
            caffe_executable: './lib/caffe_pp/build/tools/caffe'
              finetuning_dir: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/cache//cnn_finetuning/caffe_reference'

Warning: Directory already exists. 
> In start (line 137)
  In run_demo (line 1) 

opts = 

  struct with fields:

                  caffe_path: './lib/caffe_pp'
              liblinear_path: './lib/liblinear-2.1/matlab/'
              selsearch_path: './lib/SelectiveSearchCodeIJCV/'
                    imagedir: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/images'
                     basedir: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011'
                   cache_dir: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200'
              store_features: 0
                     cnn_dir: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cnn_finetuning/vgg19/'
                  batch_size: 11
                   crop_size: 224
                  finetuning: 1
            finetuning_iters: 10000
              finetuning_gpu: 0
               estimate_bbox: 1
              calculate_mean: 1
                  part_layer: 'pool5'
    part_layer_channel_count: 256
           no_selected_parts: 10
            no_visible_parts: 5
                  view_count: 10
                  iterations: 5
                  class_wise: 1
              feature_global: 'relu7'
                 use_flipped: 1
            use_bounding_box: 1
              pyramid_levels: 0
              rand_tr_images: -1
                rand_tr_part: 0
                 repetitions: 1
                   use_parts: 1
                 part_scales: [0.4400 0.2400]
                feature_part: 'relu7'
                  svm_params: '-q'
              parfor_workers: 2
                  use_parfor: 1
              verbose_output: 1
              imagelist_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/images.txt'
                  tr_ID_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/train_test_split.txt'
                 labels_file: '/home/duang/Documents/GraduateProject/Datasets/CUB-200-2011/CUB_200_2011/CUB_200_2011/image_class_labels.txt'
                      deploy: 'Simon_ICCV15_models/cnn_finetuning/vgg19//deploy.prototxt'
                       model: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cnn_finetuning/vgg19//model'
               mean_mat_file: 'Simon_ICCV15_models/cnn_finetuning/vgg19//mean.mat'
             mean_proto_file: 'Simon_ICCV15_models/cnn_finetuning/vgg19//mean.binaryproto'
            caffe_executable: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/lib/caffe_pp/build/tools/caffe'
              finetuning_dir: 'Simon_ICCV15_models/cache_iccv_cub200/cnn_finetuning/vgg19'
               part_loc_file: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200/part_locs.mat'
               est_bbox_file: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200/est_bbox.txt'
     caffe_window_file_train: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200/windows_train.txt'
       caffe_window_file_val: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200/windows_val.txt'
            caffe_part_model: '/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200/part_model.mat'

Loading part locs from /home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/Simon_ICCV15_models/cache_iccv_cub200/part_locs.mat
Loading part model from cache...
Loading estimated bounding boxes and region proposals from disk...
Using pretrained model...
Starting classification...
Warning: File:
/home/duang/Documents/GraduateProject/Codes/Constellations/part_constellation_models/part_based_classification/part_box_classification_multiscale.m
Line: 143 Column: 30
The temporary variable cur_box will be cleared at the beginning of each
iteration of the parfor loop.
Any value assigned to it before the loop will be lost.  If cur_box is used
before it is assigned in the parfor loop, a runtime error will occur.
See Parallel for Loops in MATLAB, "Temporary Variables". 
> In start (line 212)
  In run_demo (line 1) 

Error using vertcat
Dimensions of matrices being concatenated are not consistent.

Error in part_box_classification_multiscale (line 86)
        labels = [labels;labels(tr_ID)];

Error in start (line 212)
part_box_classification_multiscale( channel_ids(1:opts.no_selected_parts),
part_locs, opts );

Error in run_demo (line 1)
start('cache_dir', 'Simon_ICCV15_models/cache_iccv_cub200', ...
```
I ran the code again and printed the dimensions of some key params:
```
  Name           Size            Bytes  Class      Attributes

  tr_ID      11788x2             23576  logical              

  Name            Size             Bytes  Class     Attributes

  labels      11788x2             188608  double              

  Name               Size             Bytes  Class     Attributes

  labels(tr_ID)      17782x1             142256  double              
```
I wonder if you used the original CUB-200-2011 dataset, or maybe you have modified the format of the txt files in the dataset. 
I mean, for example, the first column of the original CUB-200-2011 dataset's txt files are the line numbers, the second column are the values. Maybe you deleted the first column and just left the second column.

**Comment by daa233** [comment]
Oh, that's right. The README has shown that your dataset's txt files have only one column (the values), which the first columns (the ids) are removed compared to the [original CUB-200-2011 dataset's txt files](http://www.vision.caltech.edu/visipedia/CUB-200-2011.html).
imagelist.txt

```
/path/to/image1.jpg
/path/to/image2.jpg
/path/to/image3.jpg
/path/to/image4.jpg
/path/to/image5.jpg
...
```

labels.txt

```
1
1
1
2
2
2
...
```

tr_ID.txt

```
0
1
1
0
1
1
...
```

**Comment by MarcelSimon** [comment]
Glad you already found the solution :)
